packages/tempus/src/Tempus_StepperDIRK_impl.hpp

```diff
@@ -230,8 +227,11 @@ void StepperDIRK<Scalar>::takeStep(
     bool pass = true;
     Thyra::SolveStatus<Scalar> sStatus;
     for (int i=0; i < numStages; ++i) {
-      if (!Teuchos::is_null(stepperDIRKObserver_))
-        stepperDIRKObserver_->observeBeginStage(solutionHistory, *this);
+        this->stepperObserver_->observeBeginStage(solutionHistory, *this);
+
+        // ???: is it a good idea to leave this (no-op) here?
+        this->stepperObserver_
+            ->observeBeforeImplicitExplicitly(solutionHistory, *this);
```

Comment:
This seems odd to have two observer calls back-to-back. Is there a reason we can not combine these?

Reply:
I just wanted to be consistent between the different RK steppers. For ERK and DIRK, some are just a nop methods. Should I just remove it?

Reply:
This is the problem with trying to make it consistent between different steppers.  I would leave them in as the other option of leaving them out would cause the issue of 'why doesn't it execute my observer?'.   Can you document that certain observers execute 'back-to-back' without any intermediate code?  Hopefully this would help alleviate some confusion.

Reply:
I will do this.

Reply:
I've updated the documentation in the RK stepper observer composite class.